app/models/manageiq/providers/ovirt/infra_manager/iso_datastore.rb

```diff
@@ -0,0 +1,54 @@
+class ManageIQ::Providers::Ovirt::InfraManager::IsoDatastore < ManageIQ::Providers::Ovirt::InfraManager::Storage
+  supports :iso_datastore
+
+  belongs_to :ext_management_system, :foreign_key => :ems_id
+
+  virtual_delegate :name, :to => :ext_management_system, :allow_nil => true, :type => :string
+
+  # Synchronize advertised images as a queued task. The
+  # queue name and the queue zone are derived from the EMS.
+  #
+  def synchronize_advertised_images_queue
+    MiqQueue.put_unless_exists(
+      :class_name  => self.class.name,
+      :instance_id => id,
+      :method_name => "synchronize_advertised_images",
+      :queue_name  => ext_management_system.queue_name_for_ems_operations,
+      :zone        => ext_management_system.try(:my_zone),
+      :role        => "ems_operations"
+    )
+  end
+
+  def advertised_images
+    return [] unless ext_management_system.kind_of?(ManageIQ::Providers::Ovirt::InfraManager)
+
+    ext_management_system.ovirt_services.advertised_images
+  end
+
+  def synchronize_advertised_images
+    log_for = "ISO Datastore on Management System <#{name}>"
+
+    _log.info("Synchronizing images on #{log_for}...")
+    db_image_hash = iso_images.index_by(&:name)
+
+    advertised_images.each do |image_name|
+      if db_image_hash.include?(image_name)
+        db_image_hash.delete(image_name)
+      else
+        iso_images.create(:name => image_name)
+      end
+    end
+
+    db_image_hash.each_value(&:destroy)
+
+    clear_association_cache
+    update_attribute(:last_refresh_on, Time.now.utc)
+
+    _log.info("Synchronizing images on #{log_for}...Complete")
+  rescue ManageIQ::Providers::Ovirt::InfraManager::OvirtServices::Error
+  end
```

Comment:
Same comment here actually, if we move the `iso_images` mapping into refresh this can all go away